packages/client/src/controllers/engine.ts

```diff
@@ -187,45 +202,51 @@ export class Engine extends IEngine {
   };
 
   public update: IEngine["update"] = async params => {
-    this.isValidUpdate(params);
+    this.isInitialized();
+    await this.isValidUpdate(params);
     const { topic, namespaces } = params;
     const id = await this.sendRequest(topic, "wc_sessionUpdate", { namespaces });
-    const { done, resolve, reject } = createDelayedPromise<void>();
+    const { done: acknowledged, resolve, reject } = createDelayedPromise<void>();
     this.events.once(engineEvent("session_update", id), ({ error }) => {
       if (error) reject(error);
       else resolve();
     });
-    await done();
     await this.client.session.update(topic, { namespaces });
+
+    return { acknowledged };
   };
 
   public extend: IEngine["extend"] = async params => {
-    this.isValidExtend(params);
+    this.isInitialized();
+    await this.isValidExtend(params);
     const { topic } = params;
     const id = await this.sendRequest(topic, "wc_sessionExtend", {});
-    const { done, resolve, reject } = createDelayedPromise<void>();
+    const { done: acknowledged, resolve, reject } = createDelayedPromise<void>();
     this.events.once(engineEvent("session_extend", id), ({ error }) => {
       if (error) reject(error);
       else resolve();
     });
-    await done();
     await this.setExpiry(topic, SESSION_EXPIRY);
+
+    return { acknowledged };
   };
 
   public request: IEngine["request"] = async params => {
-    this.isValidRequest(params);
+    this.isInitialized();
+    await this.isValidRequest(params);
     const { chainId, request, topic } = params;
     const id = await this.sendRequest(topic, "wc_sessionRequest", { request, chainId });
     const { done, resolve, reject } = createDelayedPromise<JsonRpcResponse>();
-    this.events.once<"request">(engineEvent("request", id), ({ error, data }) => {
+    this.events.once<"session_request">(engineEvent("session_request", id), ({ error, data }) => {
```

Comment:
I saw this item checked in the issue:
`Ensure awaited request returns full jsonrpc response`

Can you point me to where the actual response changes (rather than simply the event enum change)?

Reply:
Yep, its here
https://github.com/WalletConnect/walletconnect-monorepo/pull/1065/files#diff-dd17e144eb99c656245e2a405b314188f3d942c7bb843f06bea9b67d66c6358cL716-R775

Will return full payload as opposed to just result

Reply:
Gotcha thanks, missed that specific change to `payload` on my first pass 👍